# Story Generation Model Evaluation Report

## 1. Objective

The purpose of this evaluation is to compare the storytelling performance of multiple Large Language Models (LLMs) under different prompt versions. The evaluation measures both generation quality and inference latency.

Models evaluated:

- Gemini 3 Flash
- Llama 3.3 70B Versatile

Prompt variants evaluated:

- Prompt Version 1 (v1)
- Prompt Version 2 (v2)

Each model–prompt combination was tested on the same set of story-generation inputs and evaluated using an LLM-as-a-Judge approach.

---

In [1]:
import pandas as pd

## 2. Evaluation Methodology

### 2.1 Dataset

The evaluation dataset consists of generated children's stories produced from a fixed set of story prompts.

For each generated story, the following information was recorded:

- Model used
- Prompt version
- Generation latency
- Story structure validation
- Quality evaluation scores

---

### 2.2 Quality Metrics

Stories were scored on a 1–10 scale across five dimensions:

| Metric | Description |
|----------|-------------|
| Coherence | Logical flow and readability of the story |
| Consistency | Narrative consistency and absence of contradictions |
| Creativity | Originality and imagination |
| Progression | Story development and pacing |
| Language | Grammar, fluency, and language quality |

---


In [2]:
df = pd.read_csv("checkpoint3.csv")
df

,run_id,model,judge_model,prompt_version,run_number,output,latency,generation_success,eval_success,has_7_days,...,has_outro,intro_is_concise,outro_is_concise,coherence,consistency,creativity,progression,language,feedback,story_input
0,ec6064ae-d5bb-467f-89f2-bb1f345c4226,llama-3.3-70b-versatile,gemini-2.5-flash,v1,0,"{\n ""intro"": {\n ""title"": ""The Brave...",3.102684,True,True,True,...,True,True,True,6.0,9.0,6.0,6.0,9.0,"The story excels in its simple, clear language...",theme:'a brave little pirate searching for a ...
1,05c53ad2-e392-4b4d-809a-1f2c7c2f5ea7,llama-3.3-70b-versatile,gemini-2.5-flash,v1,1,"{\n ""intro"": {\n ""title"": ""Introduct...",3.053400,True,True,True,...,True,True,True,6.0,9.0,6.0,5.0,9.0,"The story is highly consistent in its tone, la...",theme:'a brave little pirate searching for a ...
2,5bcfb4b3-ed9a-4f33-b864-8f9443effef9,llama-3.3-70b-versatile,gemini-2.5-flash,v2,0,"{\n ""intro"": {\n ""title"": ""The Brave...",3.362734,True,True,True,...,True,True,True,4.0,6.0,5.0,4.0,9.0,"The story uses simple, clear language and dire...",theme:'a brave little pirate searching for a ...
3,3499d75f-afbf-4296-bf81-cca233c94fe8,llama-3.3-70b-versatile,gemini-2.5-flash,v2,1,"{\n ""intro"": {\n ""title"": ""The Journ...",3.608049,True,True,True,...,True,True,True,6.0,9.0,6.0,7.0,9.0,"The story has a strong, consistent structure a...",theme:'a brave little pirate searching for a ...
4,706f095a-fdef-492a-ac1e-36402389b234,gemini-3-flash,gemini-2.5-flash,v1,0,"{\n ""intro"": {\n ""title"": ""Pip\u2019...",8.646823,True,True,True,...,True,True,True,9.0,10.0,9.0,9.0,10.0,This story is an excellent example of interact...,theme:'a brave little pirate searching for a ...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
555,62dd6613-b475-44c1-a02d-98e6cc220588,gemini-3-flash,gemini-2.5-flash,v2,3,"{\n ""intro"": {\n ""title"": ""The Shy L...",15.657207,True,True,True,...,True,True,True,9.0,10.0,8.0,10.0,10.0,This story is an excellent example of interact...,"theme: ""a shy fox trying to make its first fri..."
556,d4b63c44-4d99-488a-a2f8-abbc87aed391,llama-3.3-70b-versatile,gemini-2.5-flash,v1,3,"{\n ""intro"": {\n ""title"": ""The Journ...",2.099175,True,True,True,...,True,True,True,8.0,9.0,6.0,7.0,9.0,"The story excels in its simple, clear language...","theme: ""a shy fox trying to make its first fri..."
557,e1df88ee-8b12-4764-854c-59a6a74e1847,llama-3.3-70b-versatile,gemini-2.5-flash,v2,3,"{\n ""intro"": {\n ""title"": ""The Journ...",3.179449,True,True,True,...,True,True,True,7.0,9.0,6.0,8.0,9.0,The story effectively guides Max from shyness ...,"theme: ""a shy fox trying to make its first fri..."
558,c990ff6b-3d45-4032-8531-788f3350779d,gemini-3-flash,gemini-2.5-flash,v2,3,"{\n ""intro"": {\n ""title"": ""Little Fi...",18.471417,True,True,True,...,True,True,True,9.0,10.0,8.0,9.0,10.0,This story is an excellent example of interact...,"theme: ""a shy fox trying to make its first fri..."


In [8]:
df =df.drop("run_number", axis=1)
df

,run_id,model,judge_model,prompt_version,output,latency,generation_success,eval_success,has_7_days,has_intro,has_outro,intro_is_concise,outro_is_concise,coherence,consistency,creativity,progression,language,feedback,story_input
0,ec6064ae-d5bb-467f-89f2-bb1f345c4226,llama-3.3-70b-versatile,gemini-2.5-flash,v1,"{\n ""intro"": {\n ""title"": ""The Brave...",3.102684,True,True,True,True,True,True,True,6.0,9.0,6.0,6.0,9.0,"The story excels in its simple, clear language...",theme:'a brave little pirate searching for a ...
1,05c53ad2-e392-4b4d-809a-1f2c7c2f5ea7,llama-3.3-70b-versatile,gemini-2.5-flash,v1,"{\n ""intro"": {\n ""title"": ""Introduct...",3.053400,True,True,True,True,True,True,True,6.0,9.0,6.0,5.0,9.0,"The story is highly consistent in its tone, la...",theme:'a brave little pirate searching for a ...
2,5bcfb4b3-ed9a-4f33-b864-8f9443effef9,llama-3.3-70b-versatile,gemini-2.5-flash,v2,"{\n ""intro"": {\n ""title"": ""The Brave...",3.362734,True,True,True,True,True,True,True,4.0,6.0,5.0,4.0,9.0,"The story uses simple, clear language and dire...",theme:'a brave little pirate searching for a ...
3,3499d75f-afbf-4296-bf81-cca233c94fe8,llama-3.3-70b-versatile,gemini-2.5-flash,v2,"{\n ""intro"": {\n ""title"": ""The Journ...",3.608049,True,True,True,True,True,True,True,6.0,9.0,6.0,7.0,9.0,"The story has a strong, consistent structure a...",theme:'a brave little pirate searching for a ...
4,706f095a-fdef-492a-ac1e-36402389b234,gemini-3-flash,gemini-2.5-flash,v1,"{\n ""intro"": {\n ""title"": ""Pip\u2019...",8.646823,True,True,True,True,True,True,True,9.0,10.0,9.0,9.0,10.0,This story is an excellent example of interact...,theme:'a brave little pirate searching for a ...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
555,62dd6613-b475-44c1-a02d-98e6cc220588,gemini-3-flash,gemini-2.5-flash,v2,"{\n ""intro"": {\n ""title"": ""The Shy L...",15.657207,True,True,True,True,True,True,True,9.0,10.0,8.0,10.0,10.0,This story is an excellent example of interact...,"theme: ""a shy fox trying to make its first fri..."
556,d4b63c44-4d99-488a-a2f8-abbc87aed391,llama-3.3-70b-versatile,gemini-2.5-flash,v1,"{\n ""intro"": {\n ""title"": ""The Journ...",2.099175,True,True,True,True,True,True,True,8.0,9.0,6.0,7.0,9.0,"The story excels in its simple, clear language...","theme: ""a shy fox trying to make its first fri..."
557,e1df88ee-8b12-4764-854c-59a6a74e1847,llama-3.3-70b-versatile,gemini-2.5-flash,v2,"{\n ""intro"": {\n ""title"": ""The Journ...",3.179449,True,True,True,True,True,True,True,7.0,9.0,6.0,8.0,9.0,The story effectively guides Max from shyness ...,"theme: ""a shy fox trying to make its first fri..."
558,c990ff6b-3d45-4032-8531-788f3350779d,gemini-3-flash,gemini-2.5-flash,v2,"{\n ""intro"": {\n ""title"": ""Little Fi...",18.471417,True,True,True,True,True,True,True,9.0,10.0,8.0,9.0,10.0,This story is an excellent example of interact...,"theme: ""a shy fox trying to make its first fri..."


### 2.3 Composite Score

A weighted overall score was calculated:

\[
TotalScore =
0.30 (Coherence) 
+0.25 (Consistency) 
+0.20 (Creativity) 
+0.15 (Progression) 
+0.10 (Language) 
\]

### Why a Weighted Score Was Used

Story quality is a multidimensional concept that cannot be accurately represented by a single evaluation criterion. While metrics such as coherence, creativity, consistency, progression, and language quality are all important, they do not contribute equally to the overall quality of a children's story.

A weighted scoring approach was therefore adopted to reflect the relative importance of each storytelling dimension:

- **Coherence (30%)** received the highest weight because a story must remain understandable and logically connected for readers to follow the narrative.
  
- **Consistency (25%)** was heavily weighted since maintaining character behavior, facts, and narrative details throughout the story is essential for story quality.

- **Creativity (20%)** was assigned a substantial weight because engaging and imaginative content is a key characteristic of children's storytelling.

- **Progression (15%)** measures how effectively the story develops and moves toward a conclusion.

- **Language Quality (10%)** was given the lowest weight because all evaluated models generally produced grammatically correct text, making it a less discriminative factor than narrative quality.

The weighted score provides a single aggregate measure that captures overall storytelling performance while prioritizing the aspects that have the greatest impact on reader experience. This enables a more meaningful comparison between models and prompt variants than treating all evaluation dimensions as equally important.   

The weighting scheme was designed based on the objectives of children's story generation, where narrative coherence and consistency are considered more critical than surface-level language quality.

---

In [9]:
df['total_score'] = ( df["coherence"] * 0.30 +
                      df["consistency"] * 0.25 +
                      df["creativity"] * 0.20 +
                      df["progression"] * 0.15 +
                      df["language"] * 0.10
                    )
df.head(10)

,run_id,model,judge_model,prompt_version,output,latency,generation_success,eval_success,has_7_days,has_intro,...,intro_is_concise,outro_is_concise,coherence,consistency,creativity,progression,language,feedback,story_input,total_score
0,ec6064ae-d5bb-467f-89f2-bb1f345c4226,llama-3.3-70b-versatile,gemini-2.5-flash,v1,"{\n ""intro"": {\n ""title"": ""The Brave...",3.102684,True,True,True,True,...,True,True,6.0,9.0,6.0,6.0,9.0,"The story excels in its simple, clear language...",theme:'a brave little pirate searching for a ...,7.05
1,05c53ad2-e392-4b4d-809a-1f2c7c2f5ea7,llama-3.3-70b-versatile,gemini-2.5-flash,v1,"{\n ""intro"": {\n ""title"": ""Introduct...",3.053400,True,True,True,True,...,True,True,6.0,9.0,6.0,5.0,9.0,"The story is highly consistent in its tone, la...",theme:'a brave little pirate searching for a ...,6.90
2,5bcfb4b3-ed9a-4f33-b864-8f9443effef9,llama-3.3-70b-versatile,gemini-2.5-flash,v2,"{\n ""intro"": {\n ""title"": ""The Brave...",3.362734,True,True,True,True,...,True,True,4.0,6.0,5.0,4.0,9.0,"The story uses simple, clear language and dire...",theme:'a brave little pirate searching for a ...,5.20
3,3499d75f-afbf-4296-bf81-cca233c94fe8,llama-3.3-70b-versatile,gemini-2.5-flash,v2,"{\n ""intro"": {\n ""title"": ""The Journ...",3.608049,True,True,True,True,...,True,True,6.0,9.0,6.0,7.0,9.0,"The story has a strong, consistent structure a...",theme:'a brave little pirate searching for a ...,7.20
4,706f095a-fdef-492a-ac1e-36402389b234,gemini-3-flash,gemini-2.5-flash,v1,"{\n ""intro"": {\n ""title"": ""Pip\u2019...",8.646823,True,True,True,True,...,True,True,9.0,10.0,9.0,9.0,10.0,This story is an excellent example of interact...,theme:'a brave little pirate searching for a ...,9.35
5,b372ca23-71df-468a-800b-53b2855e6d2f,gemini-3-flash,gemini-2.5-flash,v1,"{\n ""intro"": {\n ""title"": ""Pip\u2019...",9.611041,True,True,True,True,...,True,True,9.0,10.0,8.0,9.0,10.0,"This story excels in its consistent structure,...",theme:'a brave little pirate searching for a ...,9.15
6,9804f57a-a4b1-44c2-af8f-8d9757eccd52,gemini-3-flash,gemini-2.5-flash,v2,"{\n ""intro"": {\n ""title"": ""A Brave B...",23.939530,True,True,True,True,...,True,True,9.0,10.0,9.0,9.0,10.0,This story is an excellent example of interact...,theme:'a brave little pirate searching for a ...,9.35
7,b2a65ece-f8f7-4a7d-87c8-5835e35c9ab2,gemini-3-flash,gemini-2.5-flash,v2,"{\n ""intro"": {\n ""title"": ""Setting S...",16.303630,True,True,True,True,...,True,True,9.0,10.0,8.0,9.0,10.0,This story is an excellent example of interact...,theme:'a brave little pirate searching for a ...,9.15
8,6babf035-3208-4d98-babf-b3020945f9ad,llama-3.3-70b-versatile,gemini-2.5-flash,v1,"{\n ""intro"": {\n ""title"": ""The Quest...",4.070981,True,True,True,True,...,True,True,8.0,9.0,5.0,9.0,10.0,"The story excels in its simple, clear language...",theme:'a brave little pirate searching for a ...,8.00
9,75635102-4327-40c6-916c-e6461711cf82,llama-3.3-70b-versatile,gemini-2.5-flash,v1,"{\n ""intro"": {\n ""title"": ""The Great...",2.546060,True,True,True,True,...,True,True,8.0,9.0,8.0,9.0,10.0,This interactive story excels in its consisten...,theme:'a brave little pirate searching for a ...,8.60


## 3. Aggregated Metrics

For each `(Model, Prompt Version)` combination, the notebook computed:

- Average latency
- Average coherence
- Average consistency
- Average creativity
- Average progression
- Average language quality
- Average weighted total score

---

In [21]:
avg_latency_per_model = df.groupby(['model','prompt_version'])['latency'].mean()
avg_latency_per_model

model                    prompt_version
gemini-3-flash           v1                11.768294
                         v2                17.416692
llama-3.3-70b-versatile  v1                 2.826023
                         v2                 4.353302
Name: latency, dtype: float64

In [12]:
avg_coherence_per_model = df.groupby(['model','prompt_version'])['coherence'].mean()
avg_coherence_per_model

model                    prompt_version
gemini-3-flash           v1                8.607143
                         v2                8.571429
llama-3.3-70b-versatile  v1                6.592857
                         v2                6.592857
Name: coherence, dtype: float64

In [13]:
avg_consistency_per_model = df.groupby(['model','prompt_version'])['consistency'].mean()
avg_consistency_per_model

model                    prompt_version
gemini-3-flash           v1                9.792857
                         v2                9.814286
llama-3.3-70b-versatile  v1                8.935714
                         v2                8.635714
Name: consistency, dtype: float64

In [14]:
avg_creativity_per_model = df.groupby(['model','prompt_version'])['creativity'].mean()
avg_creativity_per_model

model                    prompt_version
gemini-3-flash           v1                8.557143
                         v2                8.621429
llama-3.3-70b-versatile  v1                6.342857
                         v2                6.657143
Name: creativity, dtype: float64

In [15]:
avg_progression_per_model = df.groupby(['model','prompt_version'])['progression'].mean()
avg_progression_per_model

model                    prompt_version
gemini-3-flash           v1                9.028571
                         v2                9.035714
llama-3.3-70b-versatile  v1                6.892857
                         v2                7.135714
Name: progression, dtype: float64

In [16]:
avg_language_per_model = df.groupby(['model','prompt_version'])['language'].mean()
avg_language_per_model

model                    prompt_version
gemini-3-flash           v1                9.892857
                         v2                9.821429
llama-3.3-70b-versatile  v1                9.200000
                         v2                9.035714
Name: language, dtype: float64

In [18]:
avg_total_score_per_model = df.groupby(['model','prompt_version'])['total_score'].mean()
avg_total_score_per_model

model                    prompt_version
gemini-3-flash           v1                9.085357
                         v2                9.086786
llama-3.3-70b-versatile  v1                7.434286
                         v2                7.442143
Name: total_score, dtype: float64

In [25]:
cols = ["latency", "coherence","consistency", "creativity", "progression", "language", "total_score"]
df_pivot = df.groupby(["model", "prompt_version","story_input"])[cols].mean().reset_index()
df_pivot

,model,prompt_version,story_input,latency,coherence,consistency,creativity,progression,language,total_score
0,gemini-3-flash,v1,theme:'a brave little pirate searching for a ...,11.345777,8.50,9.75,8.35,8.95,9.75,8.9750
1,gemini-3-flash,v1,"theme: ""a shy fox trying to make its first fri...",12.562087,8.95,10.00,7.85,9.00,10.00,9.1050
2,gemini-3-flash,v1,"theme: ""a young inventor trapped inside a magi...",12.260014,8.90,9.85,8.80,9.10,9.85,9.2425
3,gemini-3-flash,v1,theme: 'a diver searching for a glowing pearl ...,13.971315,8.80,9.90,8.85,9.00,10.00,9.2350
4,gemini-3-flash,v1,theme: 'a young wizard learning magic in a mys...,9.868386,8.55,9.85,9.35,9.10,9.95,9.2575
5,gemini-3-flash,v1,theme: 'an astronaut exploring a new planet fu...,12.286533,8.20,9.60,8.70,8.95,9.85,8.9275
6,gemini-3-flash,v1,"theme:'a fairy looking for a magical flower',\...",10.083949,8.35,9.60,8.00,9.10,9.85,8.8550
7,gemini-3-flash,v2,theme:'a brave little pirate searching for a ...,15.900995,8.55,9.80,8.40,8.95,9.85,9.0225
8,gemini-3-flash,v2,"theme: ""a shy fox trying to make its first fri...",21.911411,9.00,9.95,8.15,9.50,9.95,9.2375
9,gemini-3-flash,v2,"theme: ""a young inventor trapped inside a magi...",19.659833,8.65,9.85,8.85,9.10,9.70,9.1625



### Additionally, the following metrics were calculated:

- **Best Input Score**: Highest total score achieved on any story input.
- **Worst Input Score**: Lowest total score achieved on any story input.
- **Score Volatility**: Standard deviation of total scores across inputs.

Lower volatility indicates more consistent performance.

In [36]:
df_piv = df.groupby(["model", "prompt_version"]).agg(
    latency=('latency', 'mean'),
    coherence=('coherence', 'mean'),
    consistency=('consistency', 'mean'),
    creativity=('creativity', 'mean'),
    progression=('progression', 'mean'),
    language=('language', 'mean'),
    total_score=('total_score', 'mean'),
    
    best_input_score=('total_score', 'max'),
    worst_input_score=('total_score', 'min'),
    score_volatility=('total_score', 'std')
).reset_index()

df_piv

,model,prompt_version,latency,coherence,consistency,creativity,progression,language,total_score,best_input_score,worst_input_score,score_volatility
0,gemini-3-flash,v1,11.768294,8.607143,9.792857,8.557143,9.028571,9.892857,9.085357,9.55,6.70,0.430346
1,gemini-3-flash,v2,17.416692,8.571429,9.814286,8.621429,9.035714,9.821429,9.086786,9.80,7.05,0.459731
2,llama-3.3-70b-versatile,v1,2.826023,6.592857,8.935714,6.342857,6.892857,9.200000,7.434286,9.35,5.00,0.826301
3,llama-3.3-70b-versatile,v2,4.353302,6.592857,8.635714,6.657143,7.135714,9.035714,7.442143,9.35,3.85,1.080822


# 4. Results

## 4.1 Average Performance by Model and Prompt

| Model | Prompt | Latency (s) | Coherence | Consistency | Creativity | Progression | Language | Total Score |
|---------|---------|---------:|---------:|---------:|---------:|---------:|---------:|---------:|
| Gemini 3 Flash | v1 | 11.77 | 8.61 | 9.79 | 8.56 | 9.03 | 9.89 | **9.085** |
| Gemini 3 Flash | v2 | 17.42 | 8.57 | 9.81 | 8.62 | 9.04 | 9.82 | **9.087** |
| Llama 3.3 70B Versatile | v1 | 2.83 | 6.59 | 8.94 | 6.34 | 6.89 | 9.20 | **7.434** |
| Llama 3.3 70B Versatile | v2 | 4.35 | 6.59 | 8.64 | 6.66 | 7.14 | 9.04 | **7.442** |

Each configuration was evaluated on **140 generated stories**.

---

# 5. Best Overall Quality

**Gemini 3 Flash (v2)** achieved the highest overall score:

| Rank | Model | Prompt | Total Score |
|------|------|------|------:|
| 1 | Gemini 3 Flash | v2 | **9.087** |
| 2 | Gemini 3 Flash | v1 | **9.085** |
| 3 | Llama 3.3 70B | v2 | **7.442** |
| 4 | Llama 3.3 70B | v1 | **7.434** |

But the difference between Gemini v1 and v2 is negligible, indicating that both prompts perform similarly on Gemini except **v2 has higher latency almost double the latency of v1** so **Gemini 3 Flash (v1)** gains the highest overall score

---

# 6. Model Comparison

### Gemini 3 Flash

Strengths:

- Highest coherence scores
- Extremely strong consistency (~9.8)
- Strong creativity
- Excellent progression
- Highest language quality

Weaknesses:

- Higher latency

---

### Llama 3.3 70B Versatile

Strengths:

- Very fast generation
- Good language fluency
- Reasonably strong consistency

Weaknesses:

- Lower coherence
- Lower creativity
- Weaker story progression

The model produced decent and grammatically correct stories but struggled to match Gemini's narrative quality.

---

# 7. Prompt Engineering Impact

### Gemini

| Prompt | Total Score |
|----------|---------:|
| v1 | 9.085 |
| v2 | 9.087 |

Impact:

- Nearly identical performance
- Prompt v2 provides only a marginal improvement

Conclusion:

> Gemini appears robust to prompt wording changes and already performs near its quality ceiling.

---

### Llama

| Prompt | Total Score |
|----------|---------:|
| v1 | 7.434 |
| v2 | 7.442 |

Impact:

- Small improvement from v2
- Creativity and progression improved slightly
- Consistency decreased slightly

Conclusion:

> Prompt engineering has a measurable but limited effect on Llama's storytelling performance.

---

# 8. Latency Analysis

| Model | Prompt | Avg Latency (s) |
|---------|---------|---------:|
| Llama 3.3 70B v1 | 2.83 |
| Llama 3.3 70B v2 | 4.35 |
| Gemini 3 Flash v1 | 11.77 |
| Gemini 3 Flash v2 | 17.42 |

Observations:

- Llama is approximately **3–6× faster** than Gemini.
- Gemini's higher quality comes at the cost of significantly longer response times.
- Prompt v2 increases latency for both models.

---

# 9. Final Conclusion

The evaluation demonstrates a clear quality–latency tradeoff.

### Best Quality

**Gemini 3 Flash + Prompt v1**

- High overall score (9.085)
- High narrative quality
- High story quality

---

### Best Speed

**Llama 3.3 70B Versatile + Prompt v1**

- Lowest latency (2.83s)
- Acceptable narrative quality
- Suitable for real-time generation scenarios

---

# 10. Deployment and Rate Limit Considerations

In addition to quality and latency, API usage limits were considered when comparing the evaluated models.

| Model | Requests / Minute | Requests / Day |
|---------|---------|---------|
| Gemini 3 Flash | 5 RPM | 20 RPD |
| Llama 3.3 70B Versatile | 30 RPM | 1,000 RPD |

These limits have a direct impact on the scalability of a story generation system. While Gemini 3 Flash achieved the highest storytelling quality, its significantly lower request quotas restrict the number of stories that can be generated within a given time period. This can become a bottleneck for applications serving multiple users or requiring large-scale content generation.

In contrast, Llama 3.3 70B Versatile offers substantially higher throughput, supporting six times more requests per minute and fifty times more requests per day. Although its story quality scores were lower, its higher rate limits make it more suitable for high-volume deployment scenarios.

---

# Final choice

For a children's story generation system where storytelling quality is the primary KPI:  
i chose `Gemini 3 Flash + Prompt v1` as the `Primary` generation model and `Llama 3.3 70B Versatile + Prompt v1` as a `fallback` model when gemini hits the rate limit.